src/NServiceBus.Testing/Test.cs

```diff
@@ -216,19 +75,27 @@ public static TMessage CreateInstance<TMessage>()
         }
 
         /// <summary>
-        ///     Instantiate a new message of type TMessage performing the given action
-        ///     on the created message.
+        /// Instantiate a new message of type TMessage performing the given action
+        /// on the created message.
         /// </summary>
         public static TMessage CreateInstance<TMessage>(Action<TMessage> action)
         {
             return messageCreator.CreateInstance(action);
         }
 
-        [ThreadStatic]
-        static StubBus bus;
+        /// <summary>
+        /// Specify a test for a message handler specifying a callback to create
+        /// the handler and getting an instance of the bus passed in.
+        /// Useful for handlers based on constructor injection.
+        /// </summary>
+        [ObsoleteEx(
+            RemoveInVersion = "7",
+            TreatAsErrorFromVersion = "6")]
+        public static Handler<T> Handler<T>(Func<IMessageSession, T> handlerCreationCallback)
```

Comment:
the obsoleted signature should be `Handler<T> Handler<T>(Func<IBus, T> handlerCreationCallback)` not `IMessageSession`